src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -148,7 +154,38 @@ private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Diag
                 hasErrors: hasErrors)
             { WasCompilerGenerated = true };
 
-            return new BoundDeconstructionAssignmentOperator(node, lhsTuple, boundConversion, returnType);
+            return new BoundDeconstructionAssignmentOperator(node, lhsTuple, boundConversion, resultIsUsed, returnType);
+        }
+
+        private static bool IsDeconstructionResultUsed(ExpressionSyntax left)
+        {
+            var parent = left.Parent;
+            if (parent is null || parent.Kind() == SyntaxKind.ForEachVariableStatement)
```

Comment:
>parent is null [](start = 16, length = 14)

Are we building with compiler that performs efficient code gen for this condition?